sidechain/primitives/src/types/header.rs

```diff
@@ -0,0 +1,71 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::traits::Header as HeaderTrait;
+use codec::{Decode, Encode};
+use sp_core::H256;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+pub type ShardIdentifier = H256;
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, Copy)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Header {
+	/// The parent hash.
+	pub parent_hash: H256,
+
+	/// The block number.
+	pub block_number: u64,
+
+	/// The Shard id.
+	pub shard_id: ShardIdentifier,
+
+	/// The payload hash.
+	pub payload_hash: H256,
+}
+
+impl HeaderTrait for Header {
+	type ShardIdentifier = H256;
+
+	/// Get block number.
```

Comment:
Actually - do we need those comments in the implementation? They're already in the trait, I think that should suffice, otherwise we have these comments doubled in the code. What's your take on that?

Reply:
Agreed, we need the comments only in the traits, not the implementation.

Reply:
👍